### ISSUE TITLE
How to package program that doesn't have make install

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#763]
I'm currently writing a package to install a program. Its Makefile has an `all` target, but no `install` target. Should I tell Spack to build it in prefix, or build it in stage and copy the entire directory to prefix? I don't think there will be symlinks, but I'll have to check. Are there any other packages that currently do this?

**Comment by tgamblin** [comment]
`libdwarf` has this.  i think some others do too -- spack provides some functions in `llnl.util.filesystem` that you can use to copy things into place...

w.r.t. where to build, it's nice to build in the tmp stage b/c it is likely faster, but you need to watch out for RPATHs and such.  Generally using the spack compiler wrappers handles this and copying is fine.

If you build in the prefix, you need to do some sort of sanity check to make sure it worked -- spack by default assumes the build succeeded if there are any files in the prefix; you can add sanity checks for a more nuanced install.

**Comment by citibeth** [comment]
I've solved this problem in different ways for different cases:

*) googletest: I believe I have the Spack package copy the files.  This is not ideal, and works in part because there is nothing to compile (it's header only).

*) pism: I fixed the upstream's CMakeLists.txt so now there is a "make install."

Other possibilities I've thought of include:

*) Use a patch to add a "make install" to the packages' build, as appropriate.

*) Fork the package on GitHub and fix the build there.
